### PR TITLE
MRG: Fix plot_projs_topomap

### LINKED ISSUE #4612: BUG: plot_projs_topomap throws RT error for EEG layout
Here is the gist I used to reproduce the error with sample data
```
import mne
from mne.datasets import sample
from mne.preprocessing import compute_proj_ecg, compute_proj_eog

# getting some data ready
data_path = sample.data_path()
raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'

raw = mne.io.read_raw_fif(raw_fname, preload=True)
raw.set_eeg_reference()
raw.pick_types(meg=True, ecg=True, eog=True, stim=True)
projs, events = compute_proj_eog(raw, n_grad=1, n_mag=1, n_eeg=1)
print(projs)

mne.viz.plot_projs_topomap(projs)
```

Here is the trace

> Traceback (most recent call last):
> File "/home/ktavabi/miniconda3/envs/py2.7/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2881, in run_code
> exec(code_obj, self.user_global_ns, self.user_ns)
> File "", line 5, in 
> mne.viz.plot_projs_topomap(projs)
> File "/home/ktavabi/Projects/mne-python/mne/viz/topomap.py", line 264, in plot_projs_topomap
> % proj['desc'])
> RuntimeError: Cannot find a proper layout for projection Average EEG reference
 
@larsoner

### PR DESCRIPTION
**Pull request by larsoner**
Closes #4612.

Also fixes a bug with SSP interaction introduced in the `scalings` deprecation PR.

Ready for review/merge from my end.

@kambysese feel free to give it a try.

### CODE REVIEW

**Comment by larsoner** [comment]
https://5955-1301584-gh.circle-artifacts.com/0/tmp/circle-artifacts.1dNvWpW/html/auto_tutorials/plot_artifacts_correction_ssp.html

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -221,48 +230,57 @@ def plot_projs_topomap(projs, layout=None, cmap=None, sensors=True,
...
-            is_vv = l.kind.startswith('Vectorview')
-            if is_vv:
-                from ..channels.layout import _pair_grad_sensors_from_ch_names
-                grad_pairs = _pair_grad_sensors_from_ch_names(ch_names)
-                if grad_pairs:
-                    ch_names = [ch_names[i] for i in grad_pairs]
-
-            l_names = _clean_names(l.names, remove_whitespace=True)
-            idx = [l_names.index(c) for c in ch_names if c in l_names]
+        if isinstance(layout, Info):
```
Comment (agramfort) [inline mne/viz/topomap.py]: I don't like that. Please add an explicit info parameter and require that layout=None if info is not None

**Comment by kambysese** [comment]
Looks like same error...

> RuntimeError                              Traceback (most recent call last)
> /home/ktavabi/Projects/mne-python/mne/viz/topomap.pyc in plot_projs_topomap(projs, layout, cmap, sensors, colorbar, res, size, show, outlines, contours, image_interp, axes)
>     270                                    'projection %s, consider explicitly '
>     271                                    'passing a Layout or Info as the layout '
> --> 272                                    'parameter.' % proj['desc'])
>     273 
>     274         im = plot_topomap(data, pos[:, :2], vmax=None, cmap=cmap[0],
> 
> RuntimeError: Cannot find a proper layout for projection Average EEG reference, consider explicitly > passing a Layout or Info as the layout parameter.

**Comment by larsoner** [comment]
You need to do `plot_projs_topomap(projs, info)` (or soon `(projs, info=info)`)

**Comment by kambysese** [comment]
Okay. Well that works for [funloc dataset](https://github.com/LABSN/mnefun/pull/185)! but `mne.viz.plot_projs_topomap(projs, raw.info)` with snippet from #4612 still throws an error. 

> Traceback (most recent call last):
>   File "/home/ktavabi/miniconda3/envs/py2.7/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2881, in run_code
>     exec(code_obj, self.user_global_ns, self.user_ns)
>   File "<ipython-input-2-35c60a9da3c8>", line 15, in <module>
>     mne.viz.plot_projs_topomap(projs, raw.info)
>   File "/home/ktavabi/Projects/mne-python/mne/viz/topomap.py", line 243, in plot_projs_topomap
>     use_info = pick_info(layout, pick_channels(info_names, ch_names))
>   File "/home/ktavabi/Projects/mne-python/mne/io/pick.py", line 392, in pick_info
>     raise ValueError('No channels match the selection.')
> ValueError: No channels match the selection.

**Comment by larsoner** [comment]
> snippet from #4612 still throws an error.

That snippet has a flaw in that it does `raw.pick_types(eeg=False)` (implicitly) so I wouldn't really expect it to work for plotting an EEG projector. Maybe we could improve the error message at some point but that can probably be done separately. I guess we should probably remove the projectors that aren't relevant when doing `pick_types` (and I'm a bit surprised it doesn't).

**Comment by larsoner** [comment]
Comments addressed @agramfort. I snuck in a fix for Py3k `test_docstring_parameters.py`, too.

**Comment by agramfort** [comment]
works great

thx @larsoner

**Pull request merged**